samcli/commands/deploy/guided_config.py

```diff
@@ -102,7 +102,7 @@ def _save_parameter_overrides(self, cmd_names, config_env, parameter_overrides,
                 samconfig.put(cmd_names, self.section, "parameter_overrides", " ".join(_params), env=config_env)
 
     def _save_image_repositories(self, cmd_names, config_env, samconfig, image_repositories):
-        if image_repositories:
+        if image_repositories is not None:
```

Comment:
is this a bug we are fixing? or better validation, we may want to add typing to the function signature in that case too.

Reply:
This is an edge case where all companion stack repos are deleted which leaves image-repositories empty. With the original check, empty dict will not be saved to config.
Added comments explaining this.